pkg/strategy/dca2/recover.go

```diff
@@ -0,0 +1,266 @@
+package dca2
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/core"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+type queryAPI interface {
+	QueryOpenOrders(ctx context.Context, symbol string) ([]types.Order, error)
+	QueryClosedOrdersDesc(ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64) ([]types.Order, error)
+	QueryOrderTrades(ctx context.Context, q types.OrderQuery) ([]types.Trade, error)
+}
+
+func (s *Strategy) recover(ctx context.Context) error {
+	s.logger.Info("[DCA] recover")
+	queryService, ok := s.Session.Exchange.(queryAPI)
```

Comment:
you can create an interface called `RecoverApiQueryService`, and you can use this interface to make sure the exchange supports recover